client/asset/dcr/simnet_test.go

```diff
@@ -531,3 +577,62 @@ func runTest(t *testing.T, splitTx bool) {
 		t.Fatalf("error locking wallet: %v", err)
 	}
 }
+
+func TestTickets(t *testing.T) {
+	rig := newTestRig(t, func(name string, err error) {
+		tLogger.Infof("%s has reported a new block, error = %v", name, err)
+	})
+	defer rig.close(t)
+	tLogger.Info("Testing ticket methods with the alpha rig.")
+	testTickets(t, false, rig.alpha())
+	// TODO: beta wallet is spv but has no vps in its config. Add it and
+	// test here.
+	tLogger.Info("Testing ticket methods with the gamma rig.")
+	testTickets(t, true, rig.gamma())
+}
+
+func testTickets(t *testing.T, isInternal bool, ew *ExchangeWallet) {
+
+	// Test StakeStatus.
+	ss, err := ew.StakeStatus()
+	if err != nil {
+		t.Fatalf("unable to get stake status: %v", err)
+	}
+	tLogger.Info("The following are stake status before setting vsp or purchasing tickets.")
+	spew.Dump(ss)
+
+	// Test SetVSP.
+	err = ew.SetVSP(vspAddr)
+	if isInternal {
+		if err != nil {
+			t.Fatalf("unexected error setting spv for internal wallet: %v", err)
+		}
+	} else {
+		if err == nil {
+			t.Fatal("exected error setting spv for external wallet")
+		}
+	}
+
+	// Test PurchaseTickets.
+	if err := ew.Unlock(walletPassword); err != nil {
+		t.Fatalf("unable to unlock wallet: %v", err)
+	}
+	tickets, err := ew.PurchaseTickets(3)
+	if err != nil {
+		t.Fatalf("error purchasing tickets: %v", err)
+	}
+	tLogger.Infof("Purchased the following tickets: %v", tickets)
+
+	// Test SetVotingPreferences.
+	if err := ew.SetVotingPreferences(nil, nil, nil); err != nil {
+		t.Fatalf("error setting voting preferences: %v", err)
+	}
+
+	// Test StakeStatus again.
+	ss, err = ew.StakeStatus()
+	if err != nil {
+		t.Fatalf("unable to get stake status: %v", err)
+	}
```

Comment:
We should set the voting preferences for whatever is the latest deployment version, and then check that the values claimed in the `StakeStatus` match what we just set.

Reply:
Ah yes, I dont think this was possible without a lot of work until recently.

Reply:
Added but not sure how to allow for new deployments...

Also setting the `tspendPolicy` doesn't seem to be remembered? Maybe it needs an active tSpend? Or I could have implemented it wrong or it could be a bug.

Reply:
Some suggestions https://github.com/buck54321/dcrdex/compare/c3b9e76...buck54321:dcrdex:staking-methods-harness-tests